### PR TITLE
Add a github workflow that publishes to sdists to PyPI on release

### PR DESCRIPTION
**Pull request by corranwebster**
This doesn't attempt to do any sort of building, just packages up the sdist.

PyPI secrets will need to be added for the project.

### CODE REVIEW

File: .github/workflows/publish-on-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      with:
+        python-version: '3.8'
+
+    - name: Install Python packages needed for build and upload
+      run: |
+        python -m pip install build twine
+
+    - name: Build sdist and wheel
+      run: |
+        python setup.py sdist
```
Comment (jwiggins) [inline .github/workflows/publish-on-pypi.yml:27]: Shouldn't this be `python -m build --sdist`? Or `python -m build` if you want an sdist and a wheel? And if you're building both, does the `twine` invocation need to change?

**Review by jwiggins — COMMENTED** [review]
Some questions

File: .github/workflows/publish-on-pypi.yml (hunk 1 above)
Comment (corranwebster) [reply .github/workflows/publish-on-pypi.yml:27]: Quite likely.  I don't yet have deep experience with the new packaging tools. `python -m build --sdist` seems to match my intent.

Comment (jwiggins) [reply .github/workflows/publish-on-pypi.yml:27]: I'm using [this](https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html) as a reference here.

**Review by jwiggins — APPROVED** [review]
LGTM

**Comment by corranwebster** [comment]
Thanks.

Before this can work we'll need someone who has access to the repo settings (@mdickinson?) to provide the appropriate secret token for PyPI.

**Comment by mdickinson** [comment]
> Before this can work we'll need someone who has access to the repo settings (@mdickinson?) to provide the appropriate secret token for PyPI.

On it.

**Comment by mdickinson** [comment]
> On it.

Done. I'll post details on Slack.

**Pull request merged**